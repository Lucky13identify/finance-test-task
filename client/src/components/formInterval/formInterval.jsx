import Notiflix from 'notiflix';
import { useState } from 'react'; // Импортируем хук useState
import { Form, Input, Button } from './formInterval.styled';

export function FormForIntervalSet({ setIntervalChange }) {
  const [intervalValue, setIntervalValue] = useState('');

  const changeIntervalTime = (e, int) => {
    e.preventDefault();

    setIntervalChange(int * 1000);

    Notiflix.Notify.success(
      `The current update interval is every ${intervalValue} seconds.`
    );
  };

  const handleIntervalChange = e => {
    setIntervalValue(e.target.value);
  };

  return (
    <>
      <Form onSubmit={e => changeIntervalTime(e, intervalValue)}>
        <label htmlFor="interval">Set the interval in seconds.</label>
        <Input
          type="text"
          name="interval"
          id="interval"
          value={intervalValue}
          onChange={handleIntervalChange}
        />
        <Button type="submit" data-testid="set-interval-button">
          Set interval
        </Button>
      </Form>
    </>
  );
}
