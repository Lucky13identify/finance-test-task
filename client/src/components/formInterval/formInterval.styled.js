import styled from 'styled-components';

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 300px;
  gap: 15px;
  padding: 20px;
  margin-top: 40px;
  margin-left: 20px;

  border: 1px solid #0000cd;
  border-radius: 8px;

  background-color: #f5f5f5;
`;

export const Input = styled.input`
  width: 150px;
  border-radius: 5px;
  border: none;
  background-color: #dcdcdc;
`;

export const Button = styled.button`
  width: 150px;
  padding: 10px;

  font-family: inherit;
  font-weight: 700;
  font-size: 16px;

  cursor: pointer;
  border: none;
  border-radius: 4px;

  color: white;
  background-color: #0000cd;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1);

  &:focus,
  &:hover {
    background-color: #00008b;
  }
`;
