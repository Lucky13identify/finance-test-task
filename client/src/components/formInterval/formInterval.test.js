import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { FormForIntervalSet } from './formInterval';
import { store } from '../../redux/store';

const setIntervalChange = jest.fn();

describe('FormForIntervalSet Component', () => {
  it('should perform the interval change function', () => {
    render(
      <Provider store={store}>
        <FormForIntervalSet setIntervalChange={setIntervalChange} />
      </Provider>
    );

    const input = screen.getByRole('textbox');
    const submitButton = screen.getByText('Set interval');

    userEvent.type(input, '3');
    userEvent.click(submitButton);

    expect(setIntervalChange).toHaveBeenCalledWith(3000);
  });
  it('snapshot', () => {
    const list = render(
      <Provider store={store}>
        <FormForIntervalSet setIntervalChange={setIntervalChange} />
      </Provider>
    );

    expect(list).toMatchSnapshot();
  });
});
