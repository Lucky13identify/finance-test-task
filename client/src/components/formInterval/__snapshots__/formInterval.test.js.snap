// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormForIntervalSet Component snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="NotiflixNotifyWrap"
      style="width: 280px; z-index: 4001; opacity: 1; right: 10px; top: 10px;"
    >
      <div
        class="notiflix-notify notiflix-notify-success nx-with-animation nx-with-icon nx-fade   "
        id="NotiflixNotify-1"
        style="font-size: 13px; color: rgb(255, 255, 255); background: rgb(50, 198, 130); border-radius: 5px; pointer-events: all; font-family: \\"Quicksand\\", -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", Arial, \\"Noto Sans\\", sans-serif; animation-duration: 400ms;"
      >
        <svg
          class="nx-message-icon"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M20 0c11.03,0 20,8.97 20,20 0,11.03 -8.97,20 -20,20 -11.03,0 -20,-8.97 -20,-20 0,-11.03 8.97,-20 20,-20zm0 37.98c9.92,0 17.98,-8.06 17.98,-17.98 0,-9.92 -8.06,-17.98 -17.98,-17.98 -9.92,0 -17.98,8.06 -17.98,17.98 0,9.92 8.06,17.98 17.98,17.98zm-2.4 -13.29l11.52 -12.96c0.37,-0.41 1.01,-0.45 1.42,-0.08 0.42,0.37 0.46,1 0.09,1.42l-12.16 13.67c-0.19,0.22 -0.46,0.34 -0.75,0.34 -0.23,0 -0.45,-0.07 -0.63,-0.22l-7.6 -6.07c-0.43,-0.35 -0.5,-0.99 -0.16,-1.42 0.35,-0.43 0.99,-0.5 1.42,-0.16l6.85 5.48z"
              fill="rgba(0,0,0,0.2)"
            />
          </g>
        </svg>
        <span
          class="nx-message nx-with-icon"
        >
          The current update interval is every 3 seconds.
        </span>
      </div>
    </div>
    <div>
      <form
        class="sc-bdnyFh lnmdIb"
      >
        <label
          for="interval"
        >
          Set the interval in seconds.
        </label>
        <input
          class="sc-gtsqUy fwNfbw"
          id="interval"
          name="interval"
          type="text"
          value=""
        />
        <button
          class="sc-dlniIP gsDGzS"
          data-testid="set-interval-button"
          type="submit"
        >
          Set interval
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-bdnyFh lnmdIb"
    >
      <label
        for="interval"
      >
        Set the interval in seconds.
      </label>
      <input
        class="sc-gtsqUy fwNfbw"
        id="interval"
        name="interval"
        type="text"
        value=""
      />
      <button
        class="sc-dlniIP gsDGzS"
        data-testid="set-interval-button"
        type="submit"
      >
        Set interval
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
